doc/reference/kernel/threads/nothread.rst

```diff
@@ -0,0 +1,102 @@
+.. _nothread:
+
+Zephyr Without Threads
+######################
+
+Thread support is not necessary in some applications:
+
+* Bootloaders
+* Simple event-driven applications
+* Examples intended to demonstrate core functionality
+
+Thread support can be disabled in Zephyr by setting
+:option:`CONFIG_MULTITHREADING` to ``n``.  Since this configuration has
+a significant impact on Zephyr's functionality and testing of it has
+been limited, there are conditions on what can be expected to work in
+this configuration.
+
+What Can be Expected to Work
+****************************
+
+These core capabilities shall function correctly when
+:option:`CONFIG_MULTITHREADING` is disabled:
+
+* The :ref:`build system <application>`
+
+* The ability to boot the application to ``main()``
+
+* :ref:`Interrupt management <interrupts_v2>`
+
+* The system clock including :c:func:`k_uptime_get`
+
+* Timers, i.e. :c:func:`k_timer`
+
+* Non-sleeping delays e.g. :c:func:`k_busy_wait`.
+
+* Specifically identified drivers in certain subsystems, listed below.
+
+The expectations above affect selection of other features; for example
+:option:`CONFIG_SYS_CLOCK_EXISTS` cannot be set to ``n``.
+
+What Cannot be Expected to Work
+*******************************
+
+Functionality that will not work with :option:`CONFIG_MULTITHREADING`
+includes but is not limited to:
+
+* :c:func:`k_sleep()` and any other API that causes a thread to
```

Comment:
We should consider being explicit here, just like we are explicit above.

Reply:
Better would be to [make progress](https://github.com/zephyrproject-rtos/zephyr/issues/18970#issuecomment-736704763) on #18970 so we can just say "Any api that **sleeps** except if using the **no-wait** path".

Reply:
Since we don't have that as of now, I'd rather you mention that explicitly as @nashif mentions so we can move this forward while we wait for progress in #18970.

Reply:
I'm not sure if the direction given was to list all API that meets the "sleeps" criteria (which I can't do until #18970 progresses to the point where we've identified that API), or to just refine the text to match what I said.  So I did the latter.